### ISSUE TITLE
Adopt Obsidian official developer standards: eslint-plugin-obsidianmd + PR CI (currently: no linter, no PR CI)

### ISSUE THREAD

**Issue by nelsonlove** [nelsonlove/obsidian-mcp-suite#247]
Audit of the repo against Obsidian's official developer standards (plugin guidelines, sample-plugin tooling, `eslint-plugin-obsidianmd`), run 2026-08-19. Sources: [Plugin guidelines](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines), [obsidianmd/eslint-plugin](https://github.com/obsidianmd/eslint-plugin), [obsidian-sample-plugin](https://github.com/obsidianmd/obsidian-sample-plugin), [oo24 self-critique checklist](https://docs.obsidian.md/oo24/plugin).

## The two structural gaps

**1. No linting at all.** No ESLint, no Prettier, no lint script in any workspace. Obsidian ships an official ESLint plugin, **`eslint-plugin-obsidianmd`** (v0.4.1, ESLint 9 flat config), that encodes their plugin-guidelines checklist as rules, many auto-fixable. The sample plugin's `eslint.config.mts` is directly copyable:

```js
import obsidianmd from 'eslint-plugin-obsidianmd';
import globals from 'globals';
import { globalIgnores, defineConfig } from 'eslint/config';

export default defineConfig(
  globalIgnores(['node_modules', 'dist', 'esbuild.config.mjs', 'versions.json', 'main.js']),
  {
    languageOptions: {
      globals: { ...globals.browser },
      parserOptions: {
        projectService: { allowDefaultProject: ['eslint.config.mts', 'manifest.json'] },
        tsconfigRootDir: import.meta.dirname,
        extraFileExtensions: ['.json'],
      },
    },
  },
  ...obsidianmd.configs.recommended,
);
```

One deliberate override needed: disable/scope `obsidianmd/no-nodejs-modules` — the plugin's `node:net`/`node:fs` usage is legal because `manifest.json` sets `isDesktopOnly: true`.

**2. No PR/push CI.** `release.yml` runs tests only at tag time; nothing runs on PRs or pushes to `main`. With many parallel sessions merging PRs, this is the riskiest gap. The sample plugin's `lint.yml` runs build + lint on Node 20/22/24 per push/PR; ours should add `npm test` (the full workspace suite, currently ~2236 tests).

## What the lint pass will surface (found by manual scan, `packages/plugin/src`)

- **Bare global `app`** instead of `this.app` (rule `no-global-this`): `tool-runner.ts:47,57,64,115,214`, `connection-ui.ts:303`, `mcp/tools-nav.ts:55,85`, `mcp/tools-complementary.ts:277`, others.
- **35× `(app as any)`** (67 `as any` total) — mostly reaching private APIs (`app.commands` etc.). `typescript-eslint` comes with the config; [obsidian-typings](https://github.com/Fevol/obsidian-typings) could type most of these instead of casting.
- **Manual `<style>` injection** (rule `no-static-styles-assignment`): `kernel/governance/wiring.ts:866-871`, `settings-styles.ts:165`. Guidelines want a shipped `styles.css` + CSS variables. (`settings-styles.ts` header says no-styles.css is deliberate — decide whether to keep the exemption and disable the rule there, or migrate.)
- **`leaf.detach()`** at `kernel/governance/wiring.ts:946` (rule `detach-leaves`) — it's a live-unmount path, not `onunload`, so likely fine; needs an inline disable with justification.
- **4 unregistered `setTimeout`**: `skills/export-trigger.ts:32`, `kernel/write-queue.ts:111`, `kernel/skills/assets.ts:56`, `kernel/governance/wiring.ts:579` — verify each is safe across plugin unload.
- **2 `console.log`** in `main.ts:377,379` (guidelines: errors only by default).
- **32 direct `vault.adapter` uses** — many legitimate (config-dir files, base path), but each should be checked against "Vault API over Adapter API"; note edits belong in `Vault.process()` / `FileManager.processFrontMatter()`.
- Settings-tab heading rules, sentence-case UI text, command-ID rules (`settings-tab/*`, `ui/sentence-case`, `commands/*`) — not scanned manually; the linter will say.

Already clean, for the record: 0 `innerHTML`/`outerHTML` (comments explicitly forbid it), 0 `fetch` (no network calls), the single `setInterval` is correctly `registerInterval`-wrapped, 0 lookbehind regexes, `isDesktopOnly: true` correctly set.

## Smaller adoptions from the sample plugin

- **tsconfig hardening**: already `strict: true`; sample adds `noImplicitReturns`, `noFallthroughCasesInSwitch`, `noUncheckedIndexedAccess` (the last will surface a pile — separate effort).
- **`version-bump.mjs`**: `versions.json` (25 entries) is hand-maintained; the sample script syncs `manifest.json` + `versions.json` on `npm version patch`.
- **Build provenance attestation** in the release workflow (`actions/attest-build-provenance`) — nice-to-have.

## Explicitly nothing to take

- **Testing**: Obsidian has no official testing guidance and the sample plugin ships zero tests. Our 127-file node:test suite already exceeds the ecosystem standard — keep as is.
- **Submission requirements / developer policies**: matter only if the plugin ever goes to the community directory (it's private/BRAT). No telemetry, no network use — already compliant.

## Proposed sequencing

1. **PR 1 (small):** add `eslint-plugin-obsidianmd` flat config + `lint` scripts + a PR CI workflow (lint + `npm test` + build). Land the config with targeted rule-disables where the codebase legitimately deviates, so CI is green from day one.
2. **PR 2 (cleanup):** burn down the disables — global `app`, `as any` casts via obsidian-typings, style injection decision, unregistered timers, console noise, tsconfig hardening, version-bump script.

— filed from the obsidian-standards-audit session (bg job d49708eb)
